src/rpc/mempool.cpp

```diff
@@ -861,9 +861,9 @@ static RPCHelpMan submitpackage()
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
         {
             const UniValue raw_transactions = request.params[0].get_array();
-            if (raw_transactions.size() < 1 || raw_transactions.size() > MAX_PACKAGE_COUNT) {
+            if (raw_transactions.size() > MAX_PACKAGE_COUNT) {
```

Comment:
Instead of the 021ecfea062151bce0d67bf1575fe7e5e1ee22db approach, I actually think `must contain between 2 and MAX_PACKAGE_COUNT transactions` would be a more helpful interface for users

Reply:
I've thought about that approach but decided not to because:
- even though it's a helpful message, the user trying to submit a package of size 1 almost by definition means they don't understand the package topology requirements. Therefore, I think pointing that out is more helpful so the user can read up on the documentation right away. (of course, we could provide more specific error messages during package topology validation, but that's orthogonal to this PR)
- we'd be performing the same validation (min package size of 2) twice, which is a bit of a code smell imo

I don't feel super strongly about this so I'm happy to go along with your suggestion if you feel don't like the current approach after my explanation.

Reply:
I still prefer my deduplicated approach but since I'm okay with the alternative suggested here and it seems I'm alone in my view I've updated this in 1aa7403ac1506a2283c8553e3d2f645ed1cfc719 to reflect that.

> Don't we check for MAX_PACKAGE_COUNT later on as well (so also twice)? 

We do, but I think that's not the same - doing an early max array size check is helpful to protect against resource exhaustion.